src/main/java/org/rumbledb/runtime/functions/FunctionRuntimeIterator.java

```diff
@@ -46,7 +46,9 @@ public FunctionRuntimeIterator(
     public Item next() {
         if (this.hasNext) {
             this.hasNext = false;
-            return this.item;
+            FunctionItem result = ((FunctionItem) this.item).deepCopy();
+            result.populateClosureFromDynamicContext(this.currentDynamicContextForLocalExecution, getMetadata());
```

Comment:
First fix: when an inline function expression is evaluated, the closure must be populated from the dynamic context.